pyunlocbox/functions.py

```diff
@@ -830,7 +842,10 @@ def _prox(self, x, T):
         # Tight frame.
         if self.tight:
             tmp1 = self.A(x) - self.y()
-            scale = self.epsilon / np.sqrt(np.sum(tmp1 * tmp1, axis=0))
+            with np.errstate(divide='ignore', invalid='ignore'):
+                # Avoid 'division by zero' warning
+                scale = np.true_divide(self.epsilon,
```

Comment:
`from __future__ import division` is probably cleaner, so you know that all divisions are true divisions, regardless of Python 2 vs Python 3. Why don't you fear division by zero?